dask_match/tests/test_collection.py

```diff
@@ -5,64 +5,101 @@
 from dask.dataframe.utils import assert_eq
 from dask.utils import M
 
-from dask_match import ReadCSV, from_pandas, optimize, read_parquet
+from dask_match import read_csv, from_pandas, optimize, read_parquet
 
 
-def test_basic():
-    x = read_parquet("myfile.parquet", columns=("a", "b", "c"))
-    y = ReadCSV("myfile.csv", usecols=("a", "d", "e"))
+def _make_file(dir, format="parquet", df=None):
+    fn = os.path.join(str(dir), f"myfile.{format}")
+    if df is None:
+        df = pd.DataFrame({c: range(10) for c in "abcde"})
+    if format == "csv":
+        df.to_csv(fn)
+    elif format == "parquet":
+        df.to_parquet(fn)
+    else:
+        ValueError(f"{format} not a supported format")
+    return fn
+
+
+def test_basic(tmpdir):
+    fn_pq = _make_file(tmpdir, format="parquet")
+    fn_csv = _make_file(tmpdir, format="csv")
+
+    x = read_parquet(fn_pq, columns=("a", "b", "c"))
+    y = read_csv(fn_csv, usecols=("a", "d", "e"))
 
     z = x + y
-    result = z[("a", "b", "d")].sum(skipna="foo")
-    assert result.skipna == "foo"
-    assert result.operands[0].columns == ("a", "b", "d")
+    result = z[["a", "b", "d"]].sum(skipna=False)
+    assert result.operand("skipna") == False
+    assert result.operands[0].columns == ["a", "b", "d"]
 
     x + 1
     1 + x
 
 
-df = read_parquet("myfile.parquet", columns=["a", "b", "c"])
-df_bc = read_parquet("myfile.parquet", columns=["b", "c"])
+def test_mutation(tmpdir):
+    fn_pq = _make_file(tmpdir, format="parquet")
+    x = read_parquet(fn_pq, columns=("a", "b", "c")) + 1
+    assert "a" in x.columns
+    assert "b" in x.columns
+
+    # Check __delitem__
+    del x["a"]
+    assert "a" not in x.columns
+    assert "b" in x.columns
+
+    # Check __setitem__
+    x.a = x.b + 1
+    assert_eq(x.a.compute(), (x.b + 1).compute(), check_names=False)
+
+    # Check assign
+    y = x.assign(d=x.a + x.b, e=x.b + x.c)
+    assert_eq(y.d.compute(), (x.a + x.b).compute(), check_names=False)
+    assert_eq(y.e.compute(), (x.b + x.c).compute(), check_names=False)
+
+
+def df(fn):
+    return read_parquet(fn, columns=["a", "b", "c"])
```

Comment:
We should probably not rely on these in the basic test suite. Maybe it's time to separate.